parse/src/main/java/com/parse/ParsePlugins.java

```diff
@@ -178,7 +178,17 @@ InstallationId installationId() {
     File getCacheDir() {
         synchronized (lock) {
             if (cacheDir == null) {
-                cacheDir = new File(applicationContext.getCacheDir(), "com.parse");
+                /*
+                Check for old reference file before migration to v3.0.0.
+                If old reference found, copy the ref to new directory and delete the ref.
+                 */
+                File oldRef = applicationContext.getDir("Parse", Context.MODE_PRIVATE);
+                if (oldRef.exists()) {
+                    cacheDir = new File(oldRef, "com.parse");
+                    oldRef.deleteOnExit();
```

Comment:
I remember that because of the Android's JVM specifics and the application lifecycle it's not good to use `deleteOnExit()`. This will properly work on desktop JVM but on Android it's questionable. 

- https://groups.google.com/g/android-developers/c/2FOEz35k_Uc
- https://developer.android.com/reference/java/io/File#deleteOnExit()

> Note that on Android, the application lifecycle does not include VM termination, so calling this method will not ensure that files are deleted. Instead, you should use the most appropriate out of:

>  - Use a finally clause to manually invoke [delete()](https://developer.android.com/reference/java/io/File#delete()).
> - Maintain your own set of files to delete, and process it at an appropriate point in your application's lifecycle.
> - Use the Unix trick of deleting the file as soon as all readers and writers have opened it. No new readers/writers will be able to access the file, but all existing ones will still have access until the last one closes the file.